src/github/api.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use octocrab::models::issues::Comment;
+use octocrab::models::{AppId, InstallationRepositories};
+use octocrab::Octocrab;
+use secrecy::{ExposeSecret, SecretVec};
+
+use crate::github::GitHubRepositoryKey;
+
+/// Provides access to managed GitHub repositories.
+#[derive(Debug)]
+pub struct RepositoryAccess {
+    repositories: HashMap<GitHubRepositoryKey, Repository>,
```

Comment:
This should be a `RwLock<T>`, and we should listen to the `installation`/`installation_repositories` (not sure which are sent when) webhooks to update the `HashMap` when an installation is added or removed. I wouldn't want to restart the application every time we change the set of repositories bors uses.

Reply:
I considered "adding a new repository" to be a super rare event, but I agree that it would be nice if it could refresh automatically. I will add it in a future PR if you don't mind, to keep this one smaller.

Reply:
It would indeed be a super rare event, but not having to restart the instance would simplify the `sync-team` code a ton. I'm also totally ok doing it in a future PR, I agree we should keep PRs small-ish.